analytics-core/src/main/java/com/liferay/analytics/client/android/Analytics.kt

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.analytics.client.android
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.content.Context
+import com.liferay.analytics.client.android.api.impl.AnalyticsClientImpl
+import com.liferay.analytics.model.AnalyticsEventsMessage
+import io.reactivex.Flowable
+import io.reactivex.annotations.NonNull
+import io.reactivex.annotations.Nullable
+import io.reactivex.schedulers.Schedulers
+
+/**
+ * @author Igor Matos
+ */
+class Analytics private constructor(val application: Application, private val analyticsKey: String,
+	private val flushIntervalInMilliseconds: Long) {
+
+	fun send(@NonNull eventId: String, @NonNull properties: Map<String, String>, @NonNull applicationId: String) {
+
+		if (userId == null) {
+			return
+		}
+
+		val analyticsEventsMessageBuilder = AnalyticsEventsMessage.builder(analyticsKey, userId)
+
+		val eventBuilder = AnalyticsEventsMessage.Event.builder(applicationId, eventId)
+
+		eventBuilder.properties(properties)
+
+		analyticsEventsMessageBuilder.event(eventBuilder.build())
+
+		var flowable = Flowable.fromCallable({
+			ANALYTICS_CLIENT_IMPL.sendAnalytics(analyticsEventsMessageBuilder.build())
+		})
+
+		flowable.subscribeOn(Schedulers.newThread())
+			.subscribe()
+	}
+
+	init {
+		analyticsInstance?.let {
+			throw IllegalStateException("Your library was already initialized.")
+		}
+
+		analyticsInstance = this
```

Comment:
Why do we need this?

Reply:
@nhpatt this is to prevent the library to be initialized twice

Reply:
Oh, I see. But I think that variable should be as hidden as possible, to not have to do something like this:

val analyticsInstance = Analytics.instance
analyticsInstance.send(eventId, properties, APPLICATION_ID)

We could call the instance inside the send method and complain if it is not initialised and use it otherwise, right?

Reply:
Yeah, I agree. Earlier when I was fixing these issues I made this change, now the send method is static, and inside the method, it gets the analytics instance. But these highlighted lines are other things, is the opposite, isn't to complain if is not initialized, is to complain if is initialized and the developer tries to Analytics.Init(context, key) again, if not, it saves the instance. Was I clear?

Reply:
Yep, completely clear. I saw the use of the Analytics.instance and instead of adding the comment there, I did it in the analyticsInstance = this to try to understand the reason behind it :)